tools/clang/tools/dxr/dxr.cpp

```diff
@@ -195,19 +121,26 @@ int __cdecl wmain(int argc, const wchar_t **argv_) {
     CComPtr<IDxcOperationResult> pRewriteResult;
     CComPtr<IDxcBlobEncoding> pSource;
     std::wstring wName(CA2W(dxcOpts.InputFile.empty()? "" : dxcOpts.InputFile.data()));
-    if (!dxcOpts.InputFile.empty()) {
-      IFT_Data(FileMapDxcBlobEncoding::CreateForFile(wName.c_str(), &pSource), wName.c_str());
-    }
+    if (!dxcOpts.InputFile.empty())
+      ReadFileIntoBlob(dxcSupport, wName.c_str(), &pSource);
+
     CComPtr<IDxcLibrary> pLibrary;
     CComPtr<IDxcIncludeHandler> pIncludeHandler;
     IFT(dxcSupport.CreateInstance(CLSID_DxcLibrary, &pLibrary));
     IFT(pLibrary->CreateIncludeHandler(&pIncludeHandler));
     IFT(dxcSupport.CreateInstance(CLSID_DxcRewriter, &pRewriter));
+#ifdef _WIN32
     IFT(pRewriter->RewriteWithOptions(pSource, wName.c_str(),
                                       argv_, argc,
                                       nullptr, 0, pIncludeHandler,
                                       &pRewriteResult));
-
+#else
+    // Convert argv to wchar.
+    WArgV ArgV(argc, argv_);
```

Comment:
Does it make sense to have a passthrough mode so that the code using it doesn't need to be under ifdefs?

Reply:
Perhaps this suggestion was why WArgV was moved out of the non-windows block, but WinAdapter is only really intended for non-windows platforms. It shouldn't be required for the windows build. I also can't find the passthrough implementation